src/test/node.ts

```diff
@@ -51,6 +51,44 @@ describe("node", () => {
 			expect(messages.length).to.equal(1);
 			expect(messages).to.have.members(["hi"]);
 		});
+
+		it("it destroys the node stream and ends the ts-stream when the ts-stream's `write` call fails", async () => {
+			const errorMessage = "uh oh!";
+			const nodeStream = Readable.from(
+				(async function* () {
+					// This rejected promise will be passed to the stream's `write()` call, which will cause `write` itself to reject.
+					yield Promise.reject(errorMessage);
```

Comment:
I would change this to perform 'normal' writes from the Node stream, and instead create a custom (ts-stream) sink that causes a rejected write to happen. This is trivial to do with something like:

```ts
let ended: Error | undefined;
let aborted: Error | undefined;
stream.forEach(
  (chunk) => Promise.reject(new Error("write failed")), // note the Error ;)
  (endErr) => { ended = endErr },
  (abortErr) => { aborted = abortErr; }
);
```

And then check whether the stream is correctly ended (i.e. `endError` should contain that write failed error, and it should not be aborted (I think)).